### PR TITLE
AWS Helper functions

### PR DESCRIPTION
**Pull request by ggotti**
- Added helper functions into AWS Client to allow retrieval of data required for the cloud formation file

- [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [ ] Tests are added where necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/more-aws/aws-client.js [hunk 1]
```diff
@@ -31,6 +38,69 @@ function loadAWS(generator) {
     });
 }
 
+
+/**
+ * listRegions() returns a Promise, which resolves to an array of AWS region objects,
+ * with "Endpoint" and "RegionName" properties
+ */
+function listRegions() {
```
Comment (carvallegro) [inline generators/more-aws/aws-client.js]: I'd rather use a `region` parameter with a default value of `DEFAULT_REGION `. like you did in `listSubnets`

File: generators/more-aws/aws-client.js [hunk 2]
```diff
@@ -31,6 +38,69 @@ function loadAWS(generator) {
...
+            }
+        ]
+    };
+    return ec2.describeSubnets(params)
+        .promise().then(data => data.Subnets);
+}
+
+/**
+ * listVpcs(region) returns a Promise, which resolves to an array of AWS VPC objects, similar
+ * to the following
```
Comment (carvallegro) [inline generators/more-aws/aws-client.js]: use @param to define region. Same for all of your JSDoc comments

File: generators/more-aws/aws-client.js [hunk 3]
```diff
@@ -31,6 +38,69 @@ function loadAWS(generator) {
...
+    const ec2 = new AWS.EC2({ region });
+    return ec2.describeVpcs({})
+        .promise()
+        .then(data => data.Vpcs);
+}
+
+/**
+ * stackExists(region,stackName) checks whether the stack exists in the region
+ */
+function stackExists(region = DEFAULT_REGION, StackName) {
```
Comment (carvallegro) [inline generators/more-aws/aws-client.js]: camelCase for `StackName `?

File: generators/more-aws/index.js [hunk 4]
```diff
@@ -33,23 +32,25 @@ module.exports = class extends BaseGenerator {
                 const done = this.async();
                 awsClient.loadAWS(this)
                     .then(() => done())
-                    .catch(() => done('Error while loading the AWS library'))
+                    .catch(() => done('Error while loading the AWS library'));
```
Comment (carvallegro) [inline generators/more-aws/index.js]: If you want the generator to stop after this, the way to do is to set `this.abort = true`.
And then have `if(this.abort) return` everywhere

File: generators/more-aws/index.js [hunk 5]
```diff
@@ -33,23 +32,25 @@ module.exports = class extends BaseGenerator {
...
                 }
+
                 awsClient.saveCredentialsInAWS(profile)
                     .then(() => {
                         this.log.ok(`AWS credentials for profile ${chalk.bold(profile)} found!`);
-                        done();
                     })
+                    .then(() => awsClient.listRegions())
+                    .then((regions) => { this.regions = regions; })
+                    .then(() => done())
```
Comment (carvallegro) [inline generators/more-aws/index.js]: `.then(done)` ?

File: generators/more-aws/aws-client.js (hunk 1 above)
Comment (ggotti) [reply generators/more-aws/aws-client.js]: 🌮

File: generators/more-aws/aws-client.js (hunk 2 above)
Comment (ggotti) [reply generators/more-aws/aws-client.js]: 🌮

File: generators/more-aws/aws-client.js (hunk 3 above)
Comment (ggotti) [reply generators/more-aws/aws-client.js]: 🌮

File: generators/more-aws/index.js (hunk 4 above)
Comment (ggotti) [reply generators/more-aws/index.js]: This was just a linting fix

File: generators/more-aws/index.js (hunk 5 above)
Comment (ggotti) [reply generators/more-aws/index.js]: 🌮

**Pull request merged**